hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/recordcache/RecordCacheImpl.java

```diff
@@ -188,6 +193,10 @@ public void add(
             addToInMemoryCache(nodeId, payerAccountId, rec);
             queue.add(new TransactionRecordEntry(nodeId, payerAccountId, rec));
         }
+
+        if (states instanceof CommittableWritableStates committable) {
+            committable.commit();
+        }
```

Comment:
Instead of this, can we not let CommittableWritableStates extend from WritableStates and update the HederaState interface to return a ComittableWritableStates? Then we don't need the cast.

Reply:
We tried having CommitableWritableStates extend WritableStates.  It caused another unrelated part of the codebase, which relies on the fact that they are _not_ cross-castable, to fail.